library/internal/buffer_test.go

```diff
@@ -0,0 +1,81 @@
+package internal
+
+import (
+	"bytes"
+	"io/ioutil"
+	"sync"
+	"testing"
+)
+
+func TestMultiplexingReader_MultipleReaders(t *testing.T) {
+	buffer := MultiplexingBuffer{}
+
+	_, _ = buffer.Write([]byte("test"))
+
+	_ = buffer.Close()
+
+	reader1 := buffer.Reader()
+	reader2 := buffer.Reader()
+
+	bytes1, _ := ioutil.ReadAll(reader1)
+	bytes2, _ := ioutil.ReadAll(reader2)
+
+	if !bytes.Equal(bytes1, bytes2) {
+		t.Fatal("Reader returned different content")
+	}
+}
+
+func TestMultiplexingBuffer_StreamsData(t *testing.T) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	buffer := MultiplexingBuffer{}
+
+	reader := buffer.Reader()
+
+	var read []byte
+
+	go func() {
+		read, _ = ioutil.ReadAll(reader)
+		wg.Done()
+	}()
+
+	data := []byte("test")
+	_, _ = buffer.Write(data)
+
+	_ = buffer.Close()
```

Comment:
why error check is omitted ?

Reply:
just for simplicity when testing the functionality for myself, i wanted to add no-error checks, but forgot to change this.